internal/pkg/iface/iface.go

```diff
@@ -69,6 +69,14 @@ func FirstPublicAddress() (string, error) {
 				}
 			}
 		}
+		for _, a := range addresses {
+			// check the address type and skip if loopback
+			if ipnet, ok := a.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+				if ipnet.IP.To16() != nil && !ipnet.IP.IsLoopback() {
```

Comment:
The `!ipnet.IP.IsLoopback()` conditional is redundant, it's not possible to have `IsLoopback() == true` inside the branch. Looks like the same is true for the branch above. I believe these can be combined into one.

Reply:
I wonder if it would be ok to just do:

```go
if ipnet, ok := a.(*net.IPNet); ok && !ipnet.IP.IsLoopback() && s := ipnet.IP.String(); s != "<nil>" {
  return s, nil
}
```

The documentation says `String()` will return `<nil>` if the length would be zero.

The `String()` seems to do the equivalent of `len(ipnet.IP) == 0` to make that determination.

Reply:
Hmm. I think that it should default to IPv4 as long there is any of those addresses on any interface. However I just noticed that also my implementation handles incorrectly case where first interface have IPv6 only but others have also IPv4. Will update...